src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -452,10 +481,17 @@
         Ref("ParameterNameSegment"),
         Ref("VariableIdentifierSegment"),
     ),
+    NumericLiteralSegment=OneOf(
```

Comment:
this creates a grammar, not a literal. I'm a little bit concerned about this change, as making an integer a `integer_literal` and not a `numeric_literal` could interfere with some rules

Reply:
Ok, I think I found the real issue you mentioned and a simple solution. I have changed the type from `integer_literal` to `numeric_literal`